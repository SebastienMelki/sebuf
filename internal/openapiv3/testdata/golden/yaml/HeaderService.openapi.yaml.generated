openapi: 3.1.0
info:
    title: HeaderService API
    version: 1.0.0
paths:
    /api/v1/service-headers:
        post:
            tags:
                - HeaderService
            summary: ServiceHeadersOnly
            description: Method with no additional headers (only service headers)
            operationId: ServiceHeadersOnly
            parameters:
                - name: X-API-Key
                  in: header
                  description: API authentication key
                  required: true
                  schema:
                    type: string
                    format: uuid
                    example: 123e4567-e89b-12d3-a456-426614174000
                - name: X-Client-Version
                  in: header
                  description: Client version identifier
                  required: false
                  schema:
                    type: string
                    example: 1.2.3
            requestBody:
                content:
                    application/json:
                        schema:
                            $ref: '#/components/schemas/HeaderRequest'
                required: true
            responses:
                "200":
                    description: Successful response
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/HeaderResponse'
                "400":
                    description: Validation error
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/ValidationError'
                default:
                    description: Error response
                    content:
                        application/json:
                            schema:
                                type: object
                                properties:
                                    error:
                                        type: string
                                    code:
                                        type: integer
    /api/v1/method-headers:
        post:
            tags:
                - HeaderService
            summary: WithMethodHeaders
            description: Method with additional method-specific headers
            operationId: WithMethodHeaders
            parameters:
                - name: X-API-Key
                  in: header
                  description: API authentication key
                  required: true
                  schema:
                    type: string
                    format: uuid
                    example: 123e4567-e89b-12d3-a456-426614174000
                - name: X-Client-Version
                  in: header
                  description: Client version identifier
                  required: false
                  schema:
                    type: string
                    example: 1.2.3
                - name: X-Correlation-ID
                  in: header
                  description: Correlation ID for request tracking
                  required: false
                  schema:
                    type: string
                - name: X-Request-ID
                  in: header
                  description: Unique request identifier for tracing
                  required: true
                  schema:
                    type: string
                    format: uuid
            requestBody:
                content:
                    application/json:
                        schema:
                            $ref: '#/components/schemas/HeaderRequest'
                required: true
            responses:
                "200":
                    description: Successful response
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/HeaderResponse'
                "400":
                    description: Validation error
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/ValidationError'
                default:
                    description: Error response
                    content:
                        application/json:
                            schema:
                                type: object
                                properties:
                                    error:
                                        type: string
                                    code:
                                        type: integer
    /api/v1/override-header:
        post:
            tags:
                - HeaderService
            summary: OverrideServiceHeader
            description: Method that overrides a service header
            operationId: OverrideServiceHeader
            parameters:
                - name: X-API-Key
                  in: header
                  description: 'Override: Special API key for this method'
                  required: true
                  schema:
                    type: string
                    format: uuid
                    example: override-uuid-example
                - name: X-Client-Version
                  in: header
                  description: Client version identifier
                  required: false
                  schema:
                    type: string
                    example: 1.2.3
            requestBody:
                content:
                    application/json:
                        schema:
                            $ref: '#/components/schemas/HeaderRequest'
                required: true
            responses:
                "200":
                    description: Successful response
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/HeaderResponse'
                "400":
                    description: Validation error
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/ValidationError'
                default:
                    description: Error response
                    content:
                        application/json:
                            schema:
                                type: object
                                properties:
                                    error:
                                        type: string
                                    code:
                                        type: integer
components:
    schemas:
        FieldViolation:
            type: object
            properties:
                field:
                    type: string
                    description: The field path that failed validation (e.g., 'user.email' for nested fields). For header validation, this will be the header name (e.g., 'X-API-Key')
                description:
                    type: string
                    description: Human-readable description of the validation violation (e.g., 'must be a valid email address', 'required field missing')
            required:
                - field
                - description
            description: FieldViolation describes a single validation error for a specific field.
        ValidationError:
            type: object
            properties:
                violations:
                    type: array
                    items:
                        $ref: '#/components/schemas/FieldViolation'
                    description: List of validation violations
            required:
                - violations
            description: ValidationError is returned when request validation fails. It contains a list of field violations describing what went wrong.
        HeaderRequest:
            type: object
            properties:
                data:
                    type: string
                    description: Request data
            description: Simple request message
        HeaderResponse:
            type: object
            properties:
                result:
                    type: string
                    description: Response data
            description: Simple response message
