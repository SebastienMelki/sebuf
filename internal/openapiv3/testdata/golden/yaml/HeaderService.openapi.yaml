openapi: 3.1.0
info:
    title: HeaderService API
    version: 1.0.0
paths:
    /api/v1/service-headers:
        post:
            tags:
                - HeaderService
            summary: ServiceHeadersOnly
            description: Method with no additional headers (only service headers)
            operationId: ServiceHeadersOnly
            parameters:
                - name: X-API-Key
                  in: header
                  description: API authentication key
                  required: true
                  schema:
                    type: string
                    format: uuid
                    example: 123e4567-e89b-12d3-a456-426614174000
                - name: X-Client-Version
                  in: header
                  description: Client version identifier
                  required: false
                  schema:
                    type: string
                    example: 1.2.3
            requestBody:
                content:
                    application/json:
                        schema:
                            $ref: '#/components/schemas/HeaderRequest'
                required: true
            responses:
                "200":
                    description: Successful response
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/HeaderResponse'
                default:
                    description: Error response
                    content:
                        application/json:
                            schema:
                                type: object
                                properties:
                                    error:
                                        type: string
                                    code:
                                        type: integer
    /api/v1/method-headers:
        post:
            tags:
                - HeaderService
            summary: WithMethodHeaders
            description: Method with additional method-specific headers
            operationId: WithMethodHeaders
            parameters:
                - name: X-API-Key
                  in: header
                  description: API authentication key
                  required: true
                  schema:
                    type: string
                    format: uuid
                    example: 123e4567-e89b-12d3-a456-426614174000
                - name: X-Client-Version
                  in: header
                  description: Client version identifier
                  required: false
                  schema:
                    type: string
                    example: 1.2.3
                - name: X-Correlation-ID
                  in: header
                  description: Correlation ID for request tracking
                  required: false
                  schema:
                    type: string
                - name: X-Request-ID
                  in: header
                  description: Unique request identifier for tracing
                  required: true
                  schema:
                    type: string
                    format: uuid
            requestBody:
                content:
                    application/json:
                        schema:
                            $ref: '#/components/schemas/HeaderRequest'
                required: true
            responses:
                "200":
                    description: Successful response
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/HeaderResponse'
                default:
                    description: Error response
                    content:
                        application/json:
                            schema:
                                type: object
                                properties:
                                    error:
                                        type: string
                                    code:
                                        type: integer
    /api/v1/override-header:
        post:
            tags:
                - HeaderService
            summary: OverrideServiceHeader
            description: Method that overrides a service header
            operationId: OverrideServiceHeader
            parameters:
                - name: X-API-Key
                  in: header
                  description: 'Override: Special API key for this method'
                  required: true
                  schema:
                    type: string
                    format: uuid
                    example: override-uuid-example
                - name: X-Client-Version
                  in: header
                  description: Client version identifier
                  required: false
                  schema:
                    type: string
                    example: 1.2.3
            requestBody:
                content:
                    application/json:
                        schema:
                            $ref: '#/components/schemas/HeaderRequest'
                required: true
            responses:
                "200":
                    description: Successful response
                    content:
                        application/json:
                            schema:
                                $ref: '#/components/schemas/HeaderResponse'
                default:
                    description: Error response
                    content:
                        application/json:
                            schema:
                                type: object
                                properties:
                                    error:
                                        type: string
                                    code:
                                        type: integer
components:
    schemas:
        HeaderRequest:
            type: object
            properties:
                data:
                    type: string
                    description: Request data
            description: Simple request message
        HeaderResponse:
            type: object
            properties:
                result:
                    type: string
                    description: Response data
            description: Simple response message
